test/ethereum/network/evmScanTxMerging.test.ts

```diff
@@ -0,0 +1,256 @@
+    },
+    isSend: true,
+    memos: [],
+    nativeAmount: '0',
+    networkFee: '0',
+    otherParams: {
+      from: [abeAddress],
+      to: [bobAddress],
+      gas: '500000',
+      gasPrice: '5088740116',
+      gasUsed: '48668',
+      isFromMakeSpend: false
+    },
+    ourReceiveAddresses: [],
+    signedTx: '',
+    tokenId: null,
+    txid: '0xabcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789',
+    walletId: ''
+  }
+
+  // When the internal transaction refunds some amount back to the recipient:
+  it(`will merge internal transaction refunding`, function () {
+    const regularTx = {
+      ...templateTx,
+      isSend: true,
+      nativeAmount: '-10042',
+      networkFee: '42'
+    }
+    const internalTx = {
+      ...templateTx,
+      isSend: false,
+      nativeAmount: '100',
+      networkFee: '42',
+      ourReceiveAddresses: [bobAddress]
+    }
+    const result = mergeEdgeTransactions([regularTx, internalTx])
+    const expected = [
+      {
+        ...templateTx,
+        isSend: true,
+        nativeAmount: '-9942',
+        networkFee: '42',
+        ourReceiveAddresses: [bobAddress]
+      }
+    ]
+    assert.deepEqual(result, expected)
+  })
+
+  // When the internal transaction gives more back then sent:
+  it(`will merge internal transaction receiving of funds`, function () {
+    const regularTx = {
+      ...templateTx,
+      isSend: true,
+      nativeAmount: '-10042',
+      networkFee: '42'
+    }
+    const internalTx = {
+      ...templateTx,
+      isSend: false,
+      nativeAmount: '20042',
+      networkFee: '42',
+      ourReceiveAddresses: [bobAddress]
+    }
+    const result = mergeEdgeTransactions([regularTx, internalTx])
+    const expected = [
+      {
+        ...templateTx,
+        isSend: false,
+        nativeAmount: '10000',
+        networkFee: '42',
+        ourReceiveAddresses: [bobAddress]
+      }
+    ]
+    assert.deepEqual(result, expected)
+  })
+
+  it(`will not merge internal transaction of a different currency`, function () {
+    const regularTx = {
+      ...templateTx,
+      isSend: true,
+      nativeAmount: '-10042',
+      networkFee: '42'
+    }
+    const internalTx = {
+      ...templateTx,
+      isSend: false,
+      nativeAmount: '100',
+      networkFee: '42',
+      currencyCode: 'WBTC',
+      tokenId: '0x2260fac5e5542a773aa44fbcfedf7c193bc2c599'
+    }
+    const result = mergeEdgeTransactions([regularTx, internalTx])
+    const expected = [regularTx, internalTx]
+    assert.deepEqual(result, expected)
+  })
+
+  it(`will throw on mismatching networkFee`, function () {
+    const regularTx = {
+      ...templateTx,
+      isSend: true,
+      nativeAmount: '-10042',
+      networkFee: '42',
+      ourReceiveAddresses: [bobAddress]
+    }
+    const internalTx = {
+      ...templateTx,
+      isSend: true,
+      nativeAmount: '100',
+      networkFee: '0'
+    }
+    assert.throws(() => mergeEdgeTransactions([regularTx, internalTx]))
```

Comment:
`expect(()=> mergeEdgeTransactions(...)).throws("Failed to merge transaction 0xabcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789: Mismatch parentNetworkFee")`

The `expect` syntax is generally nicer, so we prefer it in most cases.

In this case, the `expect` syntax lets us say what error message we want. This is important, because it means the test will fail if the function starts throwing a `TypeError` or some other error besides the one we want.